channeldb/reports.go

```diff
@@ -0,0 +1,345 @@
+	// for the current chain hash.
+	ErrNoChainHashBucket = errors.New("no chain hash bucket")
+
+	// ErrNoChannelSummaries is returned when a channel is not found in the
+	// chain hash bucket.
+	ErrNoChannelSummaries = errors.New("channel bucket not found")
+)
+
+// ResolverOutcome describes the outcome for the resolver that the contract
+// court reached. This state is not necessarily final, htlcs on our own
+// commitment are resolved across two resolvers.
+type ResolverOutcome uint8
+
+const (
+	// ResolverOutcomeUnknown is a placeholder for unknown resolver
+	// outcomes.
+	ResolverOutcomeUnknown ResolverOutcome = iota
+
+	// ResolverOutcomeAnchorRecovered indicates that we swept our own
+	// anchor.
+	ResolverOutcomeAnchorRecovered
+
+	// ResolverOutcomeAnchorLost indicates that we did not sweep our anchor,
+	// or another party claimed it.
+	ResolverOutcomeAnchorLost
+
+	// ResolverOutcomeAnchorUnclaimed indicates that the anchor was left
+	// unspent.
+	ResolverOutcomeAnchorUnclaimed
+
+	// ResolverOutcomeInvalidIncomingHtlc indicates that we previously
+	// locked in an invalid htlc that we now cannot resolve.
+	ResolverOutcomeInvalidIncomingHtlc
+
+	// ResolverOutcomeIncomingHtlcSuccessTx indicates that we broadcast the
+	// first stage of a two stage htlc claim for an incoming htlc with its
+	// preimage.
+	ResolverOutcomeIncomingHtlcSuccessTx
+
+	// ResolverOutcomeIncomingHtlcTimedOut indicates that an incoming htlc
+	// timed out on chain, so we stopped waiting for the preimage.
+	ResolverOutcomeIncomingHtlcTimedOut
+
+	// ResolverOutcomeIncomingHtlcFailed indicates that our peer tried to
+	// settle an incoming htlc on chain, but we failed their attempt.
+	ResolverOutcomeIncomingHtlcFailed
+
+	// ResolverOutcomeIncomingHtlcClaimed indicates that we claimed an
+	// incoming htlc with our preimage on chain.
+	ResolverOutcomeIncomingHtlcClaimed
+
+	// ResolverTypeOutgoingHtlcTimeoutTx indicates that we broadcast the
+	// first stage of a two stage htlc timeout for an outgoing htlc.
+	ResolverTypeOutgoingHtlcTimeoutTx
+
+	// ResolverOutcomeOutgoingHtlcClaim indicates that the remote party
+	// claimed one of our outgoing htlcs on chain.
+	ResolverOutcomeOutgoingHtlcClaim
+
+	// ResolverOutcomeOutgoingHtlcTimeout indicates the we successfully
+	// timed out one of our outgoing htlcs on chain.
+	ResolverOutcomeOutgoingHtlcTimeout
+
+	// ResolverOutputCommitSweep indicates that we swept our time locked
+	// commitment output on chain.
+	ResolverOutputCommitSweep
+
+	// ResolverOutcomeCommitSweepFailed indicates that we failed to sweep
+	// out commitment output.
+	ResolverOutcomeCommitSweepFailed
+)
+
+// ResolverReport provides an account of the outcome of a resolver. This differs
+// from a ContractReport because it does not necessarily fully resolve the
+// contract; each step of two stage htlc resolution is included.
+type ResolverReport struct {
+	// OutPoint is the outpoint that was resolved.
```

Comment:
Clarify which outpoint this is exactly in case of two stage resolution? Same for amount.